pkg/cloud/metadata_test.go

```diff
@@ -394,7 +407,11 @@ func TestNewMetadataService(t *testing.T) {
 					if err == nil {
 						t.Fatalf("NewMetadataService() failed: expected error but got nothing")
 					}
-					if err.Error() != tc.getNodeErr.Error() {
+					if tc.isClientSetNil {
```

Comment:
can we add an expectedErr field to the test case struct?

The branching is hard to follow, it will be easier if every test case we simply check:

If expectedErr is non empty, did NewMetadataService() return an error?
If so, does the error match what was expected?
If not, test has failed.

I think I originally reviewed this code already and let it pass so I can help refactor .